src/ray/raylet/node_manager.cc

```diff
@@ -2928,6 +2930,79 @@ std::string compact_tag_string(const opencensus::stats::ViewDescriptor &view,
   return result.str();
 }
 
+void NodeManager::HandlePinObjectIDsRequest(const rpc::PinObjectIDsRequest &request,
+                                            rpc::PinObjectIDsReply *reply,
+                                            rpc::SendReplyCallback send_reply_callback) {
+  if (!object_pinning_enabled_) {
+    send_reply_callback(Status::OK(), nullptr, nullptr);
+    return;
+  }
+  WorkerID worker_id = WorkerID::FromBinary(request.owner_address().worker_id());
+  auto it = worker_rpc_clients_.find(worker_id);
+  if (it == worker_rpc_clients_.end()) {
+    auto client = std::unique_ptr<rpc::CoreWorkerClient>(
+        new rpc::CoreWorkerClient(request.owner_address().ip_address(),
+                                  request.owner_address().port(), client_call_manager_));
+    worker_rpc_clients_.emplace(
+        worker_id, std::make_pair<std::unique_ptr<rpc::CoreWorkerClient>, size_t>(
+                       std::move(client), 0));
+    it = worker_rpc_clients_.find(worker_id);
+  }
+  RAY_CHECK(it != worker_rpc_clients_.end());
+
+  // Pin the objects in plasma by getting them and holding a reference to
+  // the returned buffer.
```

Comment:
I would add a NOTE here that the caller must ensure that the objects are already in plasma before sending the request.